Add riposte

Jacoby6000 opened the pull request.

aardvarkk commented:
@Jacoby6000 I'm not against making the design responsive, but are you able to do it without bringing in a dependency on Bootstrap? Seems like a lot to bring in a whole UI framework just to make it behave nicer on mobile?

Jacoby6000 commented:
I just went with bootstrap because it does a lot. It's styles checkboxes and stuff to look more modern, sets up reasonable padding and margins, adds responsiveness, and bootstrap is just what everybody used back when I was doing front end work.

If there's a better way to get all this functionality without then I can do it that way. Just seems like bootstrap does a lot of work for little effort. And the assets aren't huge by today's standards.

Another thing is bootstrap has really nice plugins for things like multiselect which will be nice going forward. I imagine stat categories being better suited to a compact multiselect box with a search bar   again though, theres probably ways to get that without a whole ass framework.
________________________________
From: Ian Clarkson ***@***.***>
Sent: Monday, May 9, 2022 12:19:48 AM
To: aardvarkk/polehammer ***@***.***>
Cc: Jacob Barber ***@***.***>; Mention ***@***.***>
Subject: Re: [aardvarkk/polehammer] Responsive UI, target selection description (PR #3)


@Jacoby6000<https://github.com/Jacoby6000> I'm not against making the design responsive, but are you able to do it without bringing in a dependency on Bootstrap? Seems like a lot to bring in a whole UI framework just to make it behave nicer on mobile?

—
Reply to this email directly, view it on GitHub<https://github.com/aardvarkk/polehammer/pull/3#issuecomment-1120653734>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAUMRQBIEUCZSB6PUAWZAITVJCN7JANCNFSM5VNBH4IQ>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

aardvarkk commented:
@Jacoby6000 Alrighty -- I'll wait to see the whole setup when you're done the PR and take a peek. I often feel like I've been burned by frontend frameworks where your HTML gets so tied into the way "they" do things that you lose control of the design and it's this ultra-complex web of styling that's very difficult to navigate because you can't figure out why the heck one button doesn't line up with another or something dumb. I've used Bootstrap before (back in v3), and even trying to "upgrade" to v4 was a nightmare. Just wasn't really possible in any straightforward way. Whereas if it was all just basic CSS you've got total freedom.

Anyway, I'm also an old man about a lot of my coding style, so I'll reserve judgement until I see what it looks like when you're done with it here.

Jacoby6000 commented:
I'm still working on it a bit, but if you're curious what my current fork looks like, its available at http://dev.jacoby6000.com:3000
This branch is not all the way up to date right now. I'm trying to isolate my features to branches as well as I can.  In the latest there are additional bar charts and some min/max stuff.  I'm going to convert `categories` to multi-select next, unless some other low-hanging fruit comes my way

aardvarkk commented:
Interesting! Personally I'm not sure the bar charts add a ton to the visualization. What I *do* like about the bar charts is that they use the "actual" units. I was hoping to show that data in the chart.js Tooltip, so that when you hover over the points on the radar chart it would show you more detail about the "actual" value. I think that would kind of obviate the need for the bar charts along the bottom, but maybe you disagree. 

I also feel like the responsive UI behaves pretty strangely at the moment. I started shrinking down and got this, where the weapon names are already line-breaking but there's a ton of empty space still:

<img width="825" alt="image" src="https://user-images.githubusercontent.com/1251092/167540825-241b2d3f-5546-4b6c-a4ce-bb0d2357bc59.png">

But it's a work in progress, so we can discuss further when you feel it's all polished up.

Jacoby6000 commented:
Rationale behind the bar charts is that they provide a better visualization for direct comparisons of single stats (since they don't overlap and show things side-by-side). I question their value being shown at the same time as the radar, but I'm thinking if there were tabs, or some way to switch between radar and bar that'd be useful.

Jacoby6000 commented:
I really like the multi-select dropdown you made. And now that its condensed that way it functions way better as a set of legends.  
In my latest I've changed the graph type in to a tab selection thing.  I think it makes a lot more sense this way.   The `all` button makes way more sense with bar graphs too.

I just got mine to a place where it's "reviewable".  I can see now that bootstrap is really just providing grid-alignment.  If there's a simple way to do what I've done without bootstrap, I can work towards that. I think what's more important though is if the general feel is nice. 

Main points:
- Tabs (or other type of selection) for different graph types
- Subtext for target types
- I don't really care too much about my button styling. I'm happy to remove that
- I like the rounded borders, but again don't really care
- I also added max stats for range/speed/heavy attack/light attack.  The more I think about it the less I'm sure it's a useful metric though.
